clang/lib/Driver/ToolChains/SYCL.cpp

```diff
@@ -247,10 +254,21 @@ SYCL::getDeviceLibraries(const Compilation &C, const llvm::Triple &TargetTriple,
           C.getDriver().Diag(diag::err_drv_unsupported_option_argument)
               << A->getSpelling() << Val;
         }
-        DeviceLibLinkInfo[Val] = true && !NoDeviceLibs;
+        DeviceLibLinkInfo[Val] = !NoDeviceLibs && !IgnoreSingleLibs;
+        PrintUnusedLibWarning = IgnoreSingleLibs && !NoDeviceLibs;
       }
+      if (PrintUnusedLibWarning)
+        C.getDriver().Diag(diag::warn_ignored_clang_option)
+            << A->getSpelling() << A->getAsString(Args);
```

Comment:
Can you add a test that exercises when this warning is emitted?

Reply:
Thanks, I also noticed that there were errors with the way these flags were treated.
There's a test case now that checks if the warning is emitted and the flags are treated correctly for NVPTX:
[clang/test/Driver/sycl-device-lib-nvptx.cpp](https://github.com/intel/llvm/pull/15048/files/e574b9a3df997472af738bf9600ac39bdc76ec11#diff-cfd66e95659b53c801d104efbf6779a187ef4049935095a7a6ef1ba3e91c49cf)